### PR TITLE
Add histogram_fraction function

### PR DESCRIPTION
**Pull request by beorn7**
_This is only for the sparsehistogram branch._

@codesome early preview. I still plan to add a test similar to the test for `histogram_quantile`. (It would be much easier if we already had native histograms integrated in the PromQL testing framework, but that's still a TODO.)

### CODE REVIEW

File: promql/quantile.go [hunk 1]
```diff
@@ -186,6 +197,99 @@ func histogramQuantile(q float64, h *histogram.FloatHistogram) float64 {
...
+		}
+		if !upperSet && b.Lower >= upper {
+			upperRank = rank
+			upperSet = true
+		}
+		if lowerSet && upperSet {
+			break
+		}
+		if !lowerSet && b.Lower < lower && b.Upper > lower {
+			lowerRank = rank + (lower-b.Lower)/(b.Upper-b.Lower)*b.Count
```
Comment (codesome) [inline promql/quantile.go:271]: I am a little uncomfortable with reading this expression, this suggestion feels "safer" on a quick read although it is equivalent.
```suggestion
			lowerRank = rank + b.Count*(lower-b.Lower)/(b.Upper-b.Lower)
```

File: promql/quantile.go [hunk 2]
```diff
@@ -186,6 +197,99 @@ func histogramQuantile(q float64, h *histogram.FloatHistogram) float64 {
...
+			lowerSet = true
+		}
+		if !upperSet && b.Lower < upper && b.Upper > upper {
+			upperRank = rank + (upper-b.Lower)/(b.Upper-b.Lower)*b.Count
```
Comment (codesome) [inline promql/quantile.go:275]: Same reason as the other comment, mostly the readability.
```suggestion
			upperRank = rank + b.Count*(upper-b.Lower)/(b.Upper-b.Lower)
```

**Review by codesome — APPROVED** [review]
I am not an expert in the UI code, but they look fine. I haven't verified the test cases assuming you have done all the calculations for them. LGTM otherwise with a couple of very small nits.

**Comment by beorn7** [comment]
Addressed comments, will merge on green.

**Pull request merged**